reference/src/representation/pointers.md

```diff
@@ -0,0 +1,53 @@
+# Representation of reference and pointer types
+
+### Terminology
+
+Reference types are types of the form `&T`, `&mut T` or `&dyn T`.
+
+Raw pointer types are types of the form `*const T` or `*mut T`.
+
+### Representation
+
+The alignment of `&T`, `&mut T`, `*const T` and `*mut T` are the same,
+and are at least the word size.
+
+* If `T` is a trait, then the alignment of `&dyn T` is the word size.
+* If `T` is a sized type then the alignment of `&T` is the word size.
+* The alignment of `&[T]` is the word size.
+* The alignment of `&str` is the word size.
```

Comment:
Is the intention here to leave room for future developments? If so, it is probably saying explicitly what we are leaving room for. Perhaps something like:

---

As described above, `&T` values are typically word aligned. However, this is not guaranteed to be true for all types `T`. For example, we expect to ultimately support some form of ["Custom Dynamically Sized Types"][cdst], and in that case a type `T` may define custom metadata to be attached to a reference `&T` which could potentially affect its alignment.

[cdst]: https://internals.rust-lang.org/t/custom-dst-discussion/4842

Reply:
Is that note enough, or do we need to say more?

Reply:
Well, I didn't find it =) maybe a bullet point like:

* Alignment in other cases may vary (e.g., for other dynamically sized types).

Reply:
Done. I added a matching comment to size, since the two defns line up.